### PR TITLE
std.fs: support Lock.none on windows

### LINKED ISSUE #18430: Compilation error when attempting to lock a file (File.lock / File.tryLock
### Zig Version

0.12.0-dev.903+7aa85691b

### Steps to Reproduce and Observed Behavior

open a file:
```
var file = try dir.openFile("journal.db", .{});
defer file.close();
```

Attempt to lock the file:
```
const lock: std.fs.File.Lock = .exclusive;
const locked = try file.tryLock(lock);

```

Compilation error:
`C:\Users\henzi\bin\zig\0.12.0-dev.903+7aa85691b\files\lib\std\fs\file.zig:1546:26: error: expected type 'error{SystemResources,Unexpected,FileLocksNotSupported}!bool', found 'void'`

This seems predictable given the code in std.fs.File, specifically:
var io_status_block: windows.IO_STATUS_BLOCK = undefined;```

const exclusive = switch (l) {
    .none => return,
    .shared => false,
    .exclusive => true,
};
```
Should .none simply return false?

Sorry if this is naive!

### Expected Behavior

Assuming I am not completely off base, correct compilation.

### PR DESCRIPTION
**Pull request by notcancername**
fixes #18430

### CODE REVIEW

File: lib/std/fs/File.zig [hunk 1]
```diff
@@ -1493,7 +1493,13 @@ pub fn lock(file: File, l: Lock) LockError!void {
...
         var io_status_block: windows.IO_STATUS_BLOCK = undefined;
         const exclusive = switch (l) {
-            .none => return,
+            .none => return windows.UnlockFile(
+                file.handle,
+                &io_status_block,
+                &range_off,
+                &range_len,
+                null,
+            ) catch error.Unexpected,
```
Comment (Vexu) [inline lib/std/fs/File.zig:1502]: I don't know about the validity of the rest of this PR but this error handling is inadequate.

**Review by Vexu — CHANGES_REQUESTED** [review]

File: lib/std/fs/File.zig (hunk 1 above)
Comment (notcancername) [reply lib/std/fs/File.zig:1502]: Would you prefer `unreachable` be used here in line with `unlock`?

Comment (Vexu) [reply lib/std/fs/File.zig:1502]: You should catch the error and then switch on it. And based on other uses of the function mark both possible errors as unreachable.

Comment (notcancername) [reply lib/std/fs/File.zig]: Both errors are now unreachable, in line with `unlock`. There might be a case for returning `error.Unexpected` to the application if one doesn't consider `lock(Lock.none)` resource deallocation.

Comment (Vexu) [reply lib/std/fs/File.zig]: The same applies to the other call.

Comment (notcancername) [reply lib/std/fs/File.zig]: Of course. Thanks.

**Comment by notcancername** [comment]
CI failures seem unrelated.

**Review by Vexu — APPROVED** [review]

**Comment by andrewrk** [comment]
CI checks not passing; no update in 30+ days.

**Pull request closed without merging**